plugin.video.retrospect/resources/libs/version.py

```diff
@@ -0,0 +1,269 @@
+#===============================================================================
+# LICENSE Retrospect-Framework - CC BY-NC-ND
+#===============================================================================
+# This work is licenced under the Creative Commons 
+# Attribution-Non-Commercial-No Derivative Works 3.0 Unported License. To view a 
+# copy of this licence, visit http://creativecommons.org/licenses/by-nc-nd/3.0/ 
+# or send a letter to Creative Commons, 171 Second Street, Suite 300, 
+# San Francisco, California 94105, USA.
+#===============================================================================
+
+
+class Comparable(object):
+    """ 
+    Base Class for a simple comparison class. A derived class only needs to
+    implement an __lt__(self, other) method. All the other required methods are
+    implemented by this Comparable class and are based on the __lt__ method.
+    
+    """
+
+    def __init__(self):
+        pass
+
+    def __eq__(self, other):
+        """ Test two objects 'for Equality'
+        
+        Arguments:
+        other : Object - The other object.
+        
+        Returns:
+        True or False
+        
+        """
+
+        if other is None:
+            return False
+
+        return not self < other and not other < self
+
+    def __ne__(self, other):
+        """ Test two objects 'for non-equality'  
+        
+        Arguments:
+        other : Object - The other object.
+        
+        Returns:
+        True or False
+        
+        """
+
+        return not self.__eq__(other)
+
+    def __gt__(self, other):
+        """ Test two objects for 'greater than'
+        
+        Arguments:
+        other : Object - The other object.
+        
+        Returns:
+        True or False
+        
+        """
+
+        if other is None:
+            return True
+
+        return other < self
+
+    def __ge__(self, other):
+        """ Test two objects for 'greater or equal than'  
+        
+        Arguments:
+        other : Object - The other object.
+        
+        Returns:
+        True or False
+        
+        """
+
+        return not self < other
+
+    def __le__(self, other):
+        """ Test two objects for 'less than or equal'
+        
+        Arguments:
+        other : Object - The other object.
+        
+        Returns:
+        True or False
+        
+        """
+
+        return not other < self
+
+
+class Version(Comparable):
+    """ Class representing a version number """
```

Comment:
why?

Reply:
I have a compare-class for version numbers because I have users that copy older version of channels into new Retrospect installs, for their own reasons :(. That really breaks a lot. So I use this compare-class to see if the channels actually match the add-on version. If not, the channel is ignored.